spot-contracts/contracts/FeePolicy.sol

```diff
@@ -379,9 +361,49 @@ contract FeePolicy is IFeePolicy, OwnableUpgradeable {
     }
 
     /// @inheritdoc IFeePolicy
-    function computeDeviationRatio(SubscriptionParams memory s) public view returns (uint256) {
+    function computeDeviationRatio(SubscriptionParams memory s) public view override returns (uint256) {
         // NOTE: We assume that perp's TVL and vault's TVL values have the same base denomination.
         uint256 juniorTR = TRANCHE_RATIO_GRANULARITY - s.seniorTR;
         return (s.vaultTVL * s.seniorTR).mulDiv(ONE, (s.perpTVL * juniorTR)).mulDiv(ONE, targetSubscriptionRatio);
     }
+
+    /// @inheritdoc IFeePolicy
+    function computeRebalanceData(SubscriptionParams memory s) external view override returns (RebalanceData memory r) {
+        (bool perpDebasement, uint256 perpValueChangePerc) = computePerpEquilibriumPerc(s);
+        perpValueChangePerc = MathUpgradeable.min(
+            perpValueChangePerc,
+            (perpDebasement ? debasementValueChangePerc : enrichmentValueChangePerc)
+        );
+
+        uint256 underlyingAmtToTransfer = s.perpTVL.mulDiv(perpValueChangePerc, ONE);
+        r.protocolFeeUnderlyingAmt = underlyingAmtToTransfer.mulDiv(
+            (perpDebasement ? debasementProtocolSharePerc : enrichmentProtocolSharePerc),
+            ONE
+        );
+        underlyingAmtToTransfer -= r.protocolFeeUnderlyingAmt;
+        r.underlyingAmtIntoPerp = (perpDebasement ? int256(-1) : int256(1)) * underlyingAmtToTransfer.toInt256();
+    }
+
+    /// @notice Computes the percentage change in perpTVL (to or from the vault) to bring the system to equilibrium.
+    /// @param s The subscription parameters of both the perp and vault systems.
+    /// @return debasement True, when value has to flow from perp into the vault and False when value has to flow from the vault into perp.
+    /// @return valueChangePerc The percentage of the perpTVL that has to flow out of perp (into the vault) or into perp (from the vault)
+    ///                         to make the system perfectly balanced (to make dr = 1.0).
+    function computePerpEquilibriumPerc(
```

Comment:
Likely easier to skip calculating percentages, since this is a helper to a function that's only calculating raw balances to transfer anyway.

1) We have combined TVL
2) calc dr-normalized A (or Z)
3) figure out dr-normalized share of total TVL
4) Find difference between above and actual

Reply:
Done